### PR TITLE
add as_curve_data

### LINKED ISSUE #129: loading curve parameters from within R environment
Extend load_curve_params() to load objects in the environment, not just from OSF

> curve <- load_curve_params("https://osf.io/download/rtw5k/" )
> curve <- load_curve_params(curve) 
Error in if (file_path %>% substr(1, 4) == "http") { : 
  the condition has length > 1

This is so we can run autoplot on it (for dcm)

### PR DESCRIPTION
**Pull request by chrisorwa**

### CODE REVIEW

File: R/as_curve_data.R [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    tibble::as_tibble()
+
+  # convert data to curve class
+  converted_data <-
+    tibble_data %>% curve_converter()
+
+  return(curve_data)
+}
+
+curve_converter <- function(data, antigen_isos = NULL) {
```
Comment (d-morrison) [inline R/as_curve_data.R:26]: decomposing out `curve_converter()` seems like unnecessary complexity; there's only two commands left in `as_curve_data()`. is there a reason for doing it this way?

File: R/as_curve_data.R [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+#' curve_data <- load_curve_params("https://osf.io/download/rtw5k/")
+#'
+#' print(curve_data)
+#'
+as_curve_data <- function(data, antigen_isos = NULL) {
+
+  # convert data to a tibble data type
+  tibble_data <-
+    data %>%
+    tibble::as_tibble()
```
Comment (d-morrison) [inline R/as_curve_data.R:17]: please add a test for this code

File: R/as_curve_data.R [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#' @param data a [data.frame()] or [tibble::tbl_df]
+#' @param antigen_isos [character()] vector of antigen isotypes to be used in analyses
+#' @returns a `curve_data` object (a [tibble::tbl_df] with extra attribute `antigen_isos`)
+#' @export
+#' @examples
+#' curve_data <- load_curve_params("https://osf.io/download/rtw5k/")
+#'
+#' print(curve_data)
+#'
+as_curve_data <- function(data, antigen_isos = NULL) {
```
Comment (d-morrison) [inline R/as_curve_data.R:12]: should be called `as_curve_params()`

**Review by d-morrison — CHANGES_REQUESTED** [review]
Thanks! See comments below.

**Comment by lokune** [comment]
Closing it!

**Pull request closed without merging**